pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/IntRanges.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.core.operator.filter;
+
+import com.google.common.base.Preconditions;
+import com.linkedin.pinot.common.utils.Pairs;
+
+
+/**
+ * Various utilities to deal with integer ranges, with both bounds being inclusive (eg. for docId ranges).
+ */
+public class IntRanges {
+  /**
+   * Clips a range in place, making sure that its lower value is greater or equals to lowerBound and its upper value is
+   * smaller or equals to upperBound.
+   *
+   * @param range The range to clip
+   * @param lowerBound The lower bound
+   * @param upperBound The upper bound
+   */
+  public static void clip(Pairs.IntPair range, int lowerBound, int upperBound) {
+    range.setLeft(Math.max(range.getLeft(), lowerBound));
+    range.setRight(Math.min(range.getRight(), upperBound));
+  }
+
+  /**
+   * Checks whether a range is degenerate (upper bound lower to the lower bound)
+   *
+   * @return true if the range is degenerate
+   */
+  public static boolean isDegenerate(Pairs.IntPair range) {
+    return range.getRight() < range.getLeft();
+  }
+
+  /**
+   * Checks whether two ranges are disjoint (not overlapping or adjacent), assuming neither is degenerate.
+   *
+   * @param firstRange The first range to check
+   * @param secondRange The second range to check
+   */
+  public static boolean rangesAreDisjoint(Pairs.IntPair firstRange, Pairs.IntPair secondRange) {
+    Preconditions.checkArgument(!isDegenerate(firstRange));
+    Preconditions.checkArgument(!isDegenerate(secondRange));
+    return firstRange.getRight() < secondRange.getLeft() - 1 || secondRange.getRight() < firstRange.getLeft() - 1;
```

Comment:
I didn't follow the  -1 part. If firstRange = (3,5) and secondRange = (6,8), are you saying these are not disjoing ? Because we can collapse these ?